### ISSUE TITLE
[Feature request]Firefox search cannot be set as search in app drawer, works in latest official version, Lawnchair 2

### ISSUE THREAD

**Issue by eggsgent** [Deletescape-Media/Lawnchair#3681]
### Describe the bug

Firefox search cannot be set as search in app Dock. Regression from official version in Play store.

### Steps to reproduce

1. Have Firefox Browser installed
2. Go to Lawnchair settings and check search provider options in the Dock section

### Expected behavior

Firefox widget / search bar should be available. 

### Device information

Samsung 23, Android 14, App Version 12.1.0 beta 4

**Comment by KTibow** [comment]
> Regression from official version in Play store.

"official version" is a version thats like 5 years ago and is called Lawnchair 2

**Comment by eggsgent** [comment]
It's still a regression, and bug if a search provider is installed, bur not picked up by the launcher.

**Comment by validcube** [comment]
👋 This is not a bug, Lawnchair Launcher 12.1 don't support Firefox as Quick Search Provider!

And not a regression either! The list of search provider is hardcoded!

Checkout the list of supported search provider here!

https://github.com/LawnchairLauncher/lawnchair/blob/d97aaa35c22d0a256bbdf8d220ae5b8b626b9669/lawnchair/src/app/lawnchair/qsb/providers/QsbSearchProvider.kt#L118-L131

You can mark this issue as Feature Request.

**Comment by KTibow** [comment]
Also if Firefox is your default browser and you choose "open in browser", can't you already search with Firefox?

**Comment by eggsgent** [comment]
> And not a regression either! The list of search provider is hardcoded!

It's a regression in user experience if the previous version, and still the latest official version of the app supports it.

**Comment by KTibow** [comment]
> and still the latest official version of the app supports it.

Nope. Not "latest" - updated 3 years ago. Not as "official" as the big download button on https://lawnchair.app/. Even if it's a regression, it's not a bug. It's a feature request. Thanks for updating your issue accordingly.

**Comment by validcube** [comment]
> > And not a regression either! The list of search provider is hardcoded!
> 
> It's a regression in user experience if the previous version, and still the latest official version of the app supports it.

~~👋 Not sure about what you mean regarding the latest official version, do you mean Lawnchair v2 which is unsupported years ago?~~

~~I just recently checked and Firefox as Quick Search Provider isn't available on Lawnchair v2-2589~~

~~Would you mind giving me the version that supported Firefox so I can *hopefully* cherry-pick the change on to Lawnchair 12.1~~

### nevermind

**Comment by BeastRein** [comment]
i think the best option would be to add a list of browsers, and a separate list of search engines along with a custom text input that would allow custom search engines with the basic search engine syntax:
"https://www.google.com/search?q=%s"

this is more long term, but adding something like this would likely reduce search engine and browser requests being added in feature requests

**Comment by victor-marino** [comment]
I think we really need this.

I use Kagi as my search engine, which doesn't have any kind of app or browser. How do I use it with Lawnchair? I can't.

Sure, it could eventually be added to the list, but then somebody else will request a different search engine they use.

Why not just allow us to enter a custom search string like all web browsers do? That way all search engines are supported out of the box without the need to keep up with new ones and manually add them, which inevitably means always lagging behind.

**Comment by SuperDragonXD** [comment]
Hello. Closing this as the latest version of Lawnchair 15 now supports Firefox as the search bar widget.
